binderhub/tests/test_utils.py

```diff
@@ -0,0 +1,56 @@
+from binderhub import utils
+
+
+def test_rendezvous_rank():
+    # check that a key doesn't move if its assigned bucket remains but the
+    # other buckets are removed
+    key = "crazy frog is a crazy key"
+    first_round = utils.rendezvous_rank(["b1", "b2", "b3"], key)
+    second_round = utils.rendezvous_rank([first_round[0], first_round[1]], key)
+
+    assert first_round[0] == second_round[0], key
+
+
+def test_rendezvous_independence():
+    # check that the relative ranking of 80 buckets doesn't depend on the
+    # presence of 20 extra buckets
```

Comment:
The stability test you suggested.

Reply:
:heart: thanks! Symmetry makes it possible to phrase this as

```
    # check that the relative ranking of 80 remaining buckets doesn't depend on
    # the removal of 20 buckets from the original 100 buckets
``